lib/mdl.rb

```diff
@@ -117,10 +118,15 @@ def self.run(argv = ARGV)
     if Config[:json]
       require 'json'
       puts JSON.generate(results)
-    elsif status != 0
-      puts "\nA detailed description of the rules is available at " +
-           'https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md'
     end
     exit status
   end
+
+  private
+
+  # Creates hyperlinks in terminal emulators, if available: https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda
+  def self.linkify(text, url)
+    return text unless STDOUT.tty? && url
+    "\e]8;;#{url}\e\\#{text}\e]8;;\e\\"
```

Comment:
it might be useful to, for non-tty's to still put the url in parens after the text

Reply:
I agree; I have this a try but the default URLs are extremely long, it makes the lines quite challenging to read (as long URLs so often look like nonsense at first glance)

Is this worth a CLI option? Or adding the URL only once, before or after the block of failure lines for a specific rule?

Reply:
Printing the links out at the end in a separate block looks pretty decent — do you like this approach?

<img width="1088" alt="Screenshot 2022-01-07 at 08 03 44" src="https://user-images.githubusercontent.com/42999/148511994-7ba6afe3-a5e0-4283-b725-d726be75e03f.png">